jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -736,13 +738,32 @@ def _fit_model_to_cube(self, add_data):
         # Retrieve copy of the models with proper "fixed" dictionaries
         models_to_fit = self._reinitialize_with_fixed()
 
+        # Apply mask from selected subset
+        if self.spectral_subset_selected != "Entire Spectrum":
+            subset_mask = self.app.get_data_from_viewer(
+                self._default_spectrum_viewer_reference_name,
+                data_label=self.spectral_subset_selected
+            ).mask
+
+            if subset_mask.shape != spec.shape:
+                # broadcast to the dimensions of the full cube:
+                subset_mask = np.broadcast_to(
+                    subset_mask[None, None, :], spec.shape
+                )
+
+            if spec.mask is None:
+                spec.mask = subset_mask
+            else:
+                spec.mask += subset_mask
+
```

Comment:
if this is copy-pasted, can we put it somewhere that both spectrum and cube-fitting can use?

Reply:
It's close, but slightly different than the related passage here:

https://github.com/spacetelescope/jdaviz/blob/10fbd3bfde3694faa0e984fa5c89cef7bc820986/jdaviz/configs/default/plugins/model_fitting/model_fitting.py#L663-L672

I'll see if I can do that. 👍🏻

Reply:
Done!